rest/tests/test_learning_resource.py

```diff
@@ -0,0 +1,425 @@
+        lr_id = resource.id
+
+        vocab1_slug = self.create_vocabulary(self.repo.slug)['slug']
+        supported_term_slug = self.create_term(
+            self.repo.slug, vocab1_slug)['slug']
+
+        # This should change soon but for now we can't set this via API
+        Vocabulary.objects.get(slug=vocab1_slug).learning_resource_types.add(
+            resource.learning_resource_type
+        )
+
+        vocab_dict = dict(self.DEFAULT_VOCAB_DICT)
+        vocab_dict['name'] += " changed"
+        vocab2_slug = self.create_vocabulary(
+            self.repo.slug, vocab_dict)['slug']
+        unsupported_term_slug = self.create_term(
+            self.repo.slug, vocab2_slug)['slug']
+
+        self.assertEqual([], self.get_learning_resource(
+            self.repo.slug, lr_id)['terms'])
+
+        self.patch_learning_resource(
+            self.repo.slug, lr_id, {"terms": [supported_term_slug]})
+        self.patch_learning_resource(
+            self.repo.slug, lr_id, {"terms": ["missing"]},
+            expected_status=HTTP_400_BAD_REQUEST)
+        self.patch_learning_resource(
+            self.repo.slug, lr_id, {"terms": [unsupported_term_slug]},
+            expected_status=HTTP_400_BAD_REQUEST)
+
+    def test_learning_resource_types(self):
+        """
+        Get from learning_resource_types
+        """
+        base_url = "{}learning_resource_types/".format(API_BASE)
+
+        resp = self.client.get(base_url)
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        types = as_json(resp)
+
+        self.assertEqual(sorted([lrt.name for lrt
+                                 in LearningResourceType.objects.all()]),
+                         sorted([t['name'] for t in types['results']]))
+
+        # nothing besides GET, OPTION, HEAD allowed
+        resp = self.client.options(base_url)
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        resp = self.client.head(base_url)
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        resp = self.client.post(base_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+        resp = self.client.patch(base_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+        resp = self.client.put(base_url, {})
+        self.assertEqual(HTTP_405_METHOD_NOT_ALLOWED, resp.status_code)
+
+        # restricted to logged in users
+        self.logout()
+        resp = self.client.get(base_url)
+        self.assertEqual(HTTP_403_FORBIDDEN, resp.status_code)
+
+        # but otherwise unrestricted
+        self.login(self.user_norepo)
+        resp = self.client.get(base_url)
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        types = as_json(resp)
+
+        self.assertEqual(sorted([lrt.name for lrt
+                                 in LearningResourceType.objects.all()]),
+                         sorted([t['name'] for t in types['results']]))
+
+
+# pylint: disable=too-many-ancestors
```

Comment:
This doesn't work, I believe it needs to be before the class to have an effect on it